guide/interactions/slash-command-permissions.md

```diff
@@ -1,104 +1,221 @@
+
+		const pingCommand = response.find(cmd => cmd.name === 'ping');
+
+		const fullPermissions = [
+			{
+				id: pingCommand.id,
+				permissions: [
+					{
+						id: '1234567890',
+						type: 1,
+						permission: true,
+					},
+				],
+			},
+		];
+
+		await rest.put(
+			Routes.guildApplicationCommandsPermissions(clientId, guildId),
+			{ body: fullPermissions },
+		);
+
+		console.log('Successfully applied permission overwrite.');
+	} catch (error) {
+		console.error(error);
+	}
+})();
 ```
 :::
+::::
 
-If you have a command that is disabled by default and you want to grant someone access to use it, do as follows:
+::: warning
+We are not using the command handler version of the `deploy-commands.js` file for the sake of simplicity during explanation. However, all the concepts and examples in this page are also applicable for the command handler version of the aformentioned file.
+:::
 
-<!-- eslint-skip -->
+As you can see in line 16, calling `rest.put()` to this specific endpoint `Routes.applicationGuildCommands()` will return a response. We store that response in the variable called `response`. This response will be an array of all the registered commands.
 
-```js {5}
-const permissions = [
-	{
-		id: '224617799434108928',
-		type: 'USER',
-		permission: true,
-	},
-];
+Then, in line 23, we retrieve the `ping` command from that array with [`Array.find()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find). Once you retrieve the `ping` command, you can access its ID from the `id` property.
 
-await command.permissions.set({ permissions });
-```
+- Guild ID ✅
+- Command ID ✅
+- User/role ID
 
+Next, in line 25 to 33, you define an array of JSON objects. This JSON needs to have two properties: `id` and `permissions`. The `id` should be the command ID (line 27) and the `permissions` should be an array of [Application Command Permissions Structure](https://discord.com/developers/docs/interactions/application-commands#application-command-permissions-object-application-command-permissions-structure) objects (line 28-34). 
 
-## Role permissions
+This `Application Command Permissions Structure` object needs to have three properties: `id`, `type`, and `permission`. The `id` will be the user id or role id that you want to allowlist (line 30). The `type` specifies what type of `id` it is (line 31). Type `1` means `ROLE` id, and type `2` means `USER` id. Finally, the `permission` specifies if you want to allow or disallow this `id` to use the command (line 32). If `permission` is `true` then it means allow, if it is `false` then it means disallow. Since your `ping` command is already registered with `setDefaultPermission(false)`, it wouldn't make much sense to also have a permission overwrite to disallow specific users or roles. On top of that, you are using the allowlist approach, which means you are enabling the command for a specific role, hence you should set `permission: true`.
```

Comment:
```suggestion
This `Application Command Permissions Structure` object needs to have three properties: `id`, `type`, and `permission`. The `id` will be the user id or role id that you want to allow running the command. The `type` specifies what type of `id` it is. Type `1` is for roles, and type `2` is for users. Finally, the `permission` specifies if you want to allow or disallow this `id` to use the command. If `permission` is `true` then it means allow, if it is `false` then it means disallow. Since we want to allow users, we set `permission` to true.
```